plugins/InspectionGadgets/InspectionGadgetsAnalysis/src/com/siyeh/ig/bugs/IncorrectDateTimeFormatInspection.java

```diff
@@ -0,0 +1,262 @@
+// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.siyeh.ig.bugs;
+
+import com.intellij.codeInspection.AbstractBaseJavaLocalInspectionTool;
+import com.intellij.codeInspection.LocalQuickFix;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.psi.*;
+import com.intellij.psi.util.ConstantExpressionUtil;
+import com.siyeh.InspectionGadgetsBundle;
+import com.siyeh.ig.callMatcher.CallMatcher;
+import com.siyeh.ig.psiutils.ExpressionUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.siyeh.ig.callMatcher.CallMatcher.*;
+
+public class IncorrectDateTimeFormatInspection extends AbstractBaseJavaLocalInspectionTool {
+
+  private interface CountVerifier {
+    boolean verifier(int count);
+  }
+
+  private static class RangeVerifier implements CountVerifier {
+    private final int from;
+    private final int to;
+
+    private RangeVerifier(int from, int to) {
+      this.from = from;
+      this.to = to;
+    }
+
+    @Override
+    public boolean verifier(int count) {
+      return count >= from && count <= to;
+    }
+  }
+
+  private static RangeVerifier rangeOf(int from, int to) {
+    return new RangeVerifier(from, to);
+  }
+
+  private static class SetVerifier implements CountVerifier {
+    private final Set<Integer> set;
+
+    private SetVerifier(int... numbers) {
+      set = Arrays.stream(numbers).boxed()
```

Comment:
Probably use it.unimi.dsi.fastutil.ints.IntOpenHashSet? It's available in platform and much more efficient. Or a BitSet, taken into account current uses.

Reply:
I rewrote with `IntOpenHashSet`